catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformations/EmbeddedListMetacardsHandler.java

```diff
@@ -82,23 +91,24 @@ private void removeExternalListAttributes(Map listMetacardMap) {
   }
 
   /**
-   * Given a {@link org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceMetacardImpl}, get a list
-   * of actions that can be executed on a list.
+   * Given a {@link org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceMetacardImpl} and a {@link
+   * org.codice.ddf.catalog.ui.metacard.workspace.ListMetacardImpl}, get a list of actions that can
+   * be executed on a list.
    */
-  private List<Map<String, Object>> getListActions(Metacard workspaceMetacard) {
-    final String host = SystemBaseUrl.EXTERNAL.getBaseUrl();
+  private List<Map<String, Object>> getListActions(
+      Metacard workspaceMetacard, Metacard listMetacard) {
+    final Map<String, Metacard> listContext =
+        ImmutableMap.of("workspace", workspaceMetacard, "list", listMetacard);
 
     return actionRegistry
-        .list(workspaceMetacard)
+        .list(listContext)
         .stream()
         .filter(action -> action.getId().startsWith(LIST_ACTION_PREFIX))
         .map(
             action -> {
-              // Work-around for paths being behind VPCs with non-public DNS values
-              final String url = action.getUrl().toString().replaceFirst(host, "");
```

Comment:
❓ This is no longer necessary?

Reply:
We don't need this anymore now that we have support for internal/external URLs.